diff --git a/contrib/init/systemd/docker.service b/contrib/init/systemd/docker.service
index c3f3472..baa54ea 100644
--- a/contrib/init/systemd/docker.service
+++ b/contrib/init/systemd/docker.service
@@ -6,10 +6,11 @@ Requires=docker.socket
 
 [Service]
 Type=notify
+EnvironmentFile=-/etc/sysconfig/docker
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/dockerd -H fd://
+ExecStart=/usr/bin/dockerd $DOCKER_OPTS -H fd:// -g /var/cache/docker
 ExecReload=/bin/kill -s HUP $MAINPID
 # Having non-zero Limit*s causes performance problems due to accounting overhead
 # in the kernel. We recommend using cgroups to do container-local accounting.
