diff --git a/contrib/init/systemd/docker.service b/contrib/init/systemd/docker.service
index 75cb68c..63670c2 100644
--- a/contrib/init/systemd/docker.service
+++ b/contrib/init/systemd/docker.service
@@ -6,10 +6,11 @@ Requires=docker.socket
 
 [Service]
 Type=notify
+EnvironmentFile=-/etc/sysconfig/docker
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/docker daemon -H fd://
+ExecStart=/usr/bin/docker daemon $DOCKER_OPTS -H fd://
 MountFlags=slave
 LimitNOFILE=1048576
 LimitNPROC=1048576
