diff -up moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_common.go.2~ moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_common.go
diff -up moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_linux.go.2~ moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_linux.go
--- moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_linux.go.2~	2024-06-27 14:52:23.742996913 +0200
+++ moby-27.0.2/vendor/github.com/moby/buildkit/executor/runcexecutor/executor_linux.go	2024-06-27 14:56:28.029590351 +0200
@@ -25,9 +25,6 @@ func (w *runcExecutor) run(ctx context.C
 	killer := newRunProcKiller(w.runc, id)
 	return w.callWithIO(ctx, process, started, killer, func(ctx context.Context, started chan<- int, io runc.IO, pidfile string) error {
 		extraArgs := []string{}
-		if keep {
-			extraArgs = append(extraArgs, "--keep")
-		}
 		_, err := w.runc.Run(ctx, id, bundle, &runc.CreateOpts{
 			NoPivot:   w.noPivot,
 			Started:   started,
